### ISSUE TITLE
Column 'visit_total_interactions' is missing

### ISSUE THREAD

**Issue by billtzim** [piwik/piwik#11074]
Logs show the following error

Piwik (tracker): Error query: fetch() failed: Unknown column 'visit_total_interactions' in 'field list'.....

I see it was added recently ( https://github.com/piwik/piwik/commit/107147670f46b234afadefe82e0c384b10c41279 ) but there is no create reference for it in the Mysql Schema

**Comment by mattab** [comment]
There is no reference because the code automagically should detect it and offer to add the new field.

Do the files   plugins/Actions/Columns/InteractionPosition.php  and    plugins/Actions/Columns/VisitTotalInteractions.php  exist in your Piwik?

**Comment by mattab** [comment]
I would asume that the files are missing in your Piwik. If so, please re-upload all files from Piwik 3. 

If they are present, please re-open this issue

**Comment by billtzim** [comment]
Hello, our piwik installation is not missing the mentioned files

**Comment by tsteur** [comment]
not without investigating. would say actions plugin is disabled. But then it should also not track anything into that column

**Comment by billtzim** [comment]
I have tried to deactivate / activate the Actions and Live plugins as i have red in other posts but the behaviour remains

**Comment by tolisvv** [comment]
I added the column manually
`alter` table piwik_log_visit add column visit_total_interactions MEDIUMINT UNSIGNED DEFAULT 0;`

**Comment by mattab** [comment]
For now we can't reproduce this issue, but if it occurs again in the future we can hopefully find the root cause

**Comment by piwikoper** [comment]
Hi gentlemen
We did face a very similar issue with another column. The add column statement couldn't be executed since we did not have sufficient space on OS level to store a copy of the table:
```
mysql> alter table piwik_log_link_visit_action add column interaction_position SMALLINT UNSIGNED DEFAULT NULL;
ERROR 1114 (HY000): The table 'piwik_log_link_visit_action' is full
```
We do have the assumption that the return code of the SQL statement might not be checked properly and the failure of the execution is not detected on application level.
We did add some storage and it was working perfectly.
